lua/telescope/_extensions/file_browser/make_entry.lua

```diff
@@ -218,9 +229,11 @@ local make_entry = function(opts)
     return rawget(t, rawget({ value = 1 }, k))
   end
 
-  return function(absolute_path)
+  return function(entry)
+    local absolute_path = entry[1]
     local e = setmetatable({
       absolute_path,
+      git = entry[2],
```

Comment:
Without having looked at the rest of the PR here (for brevity most of all) - I think a cleaner pattern here is to pass a `{ absolute path = git_status }` dictionary table to whenever the `entry_maker` is first instantiated and hand over to the finder.

That way you can just put it into the `__index` method and return from the lookup, if the lookup table exists and a corresponding git status for `entry.value` exists.

Seems cleaner for cases in which git status may or may not exist. For instance, the RFC / feat PR will enable an auto switch to something akin to `telescope.builtin.find_files` which should be fully async and hence not show git status. That's done via intermittently creating a new finder. I hope that makes somewhat sense :sweat_smile:

Reply:
Makes sense, that might also make the directory case easier to implement to also mark directories which include changed files.

I am just not 100% sure how I would hand this over. Do you have an example? The `results` table was so far the only option I found to pass something to the `entry_maker`. I guess that also in line with me not fully understanding what this `__index` metamethod does.

Reply:
Oh. It's this line:
```lua
  local entry_maker = opts.entry_maker { cwd = opts.path }
```
right? I misread it because I'm not used to the trick in Lua to omitting function brackets when calling a function. I would just change it to something like
```lua
  local entry_maker = opts.entry_maker { cwd = opts.path, git_statuses = git_statuses }
```

Reply:
Just a quick 'exactly' to confirm from mobile :)